src/v0/destinations/dynamic_yield/transform.js

```diff
@@ -0,0 +1,107 @@
+const sha256 = require('sha256');
+const { EventType } = require('../../../constants');
+const { BASE_URL, ConfigCategory, mappingConfig } = require('./config');
+const {
+  defaultRequestConfig,
+  defaultPostRequestConfig,
+  simpleProcessRouterDest,
+  removeUndefinedAndNullAndEmptyValues,
+  constructPayload,
+} = require('../../util');
+const {
+  ConfigurationError,
+  TransformationError,
+  InstrumentationError,
+} = require('../../util/errorTypes');
+const { preparePayload } = require('./util');
+
+const responseBuilder = (payload, apiKey, endpoint) => {
+  if (payload) {
+    const response = defaultRequestConfig();
+    response.endpoint = endpoint;
+    response.headers = {
+      'Content-Type': 'application/json',
+      'DY-API-Key': apiKey,
+    };
+    response.method = defaultPostRequestConfig.requestMethod;
+    response.body.JSON = removeUndefinedAndNullAndEmptyValues(payload);
+    return response;
+  }
+  throw new TransformationError('Payload could not be populated due to wrong input');
+};
+
+const identifyResponseBuilder = (message, { Config }) => {
+  const { apiKey } = Config;
+
+  const payload = constructPayload(message, mappingConfig[ConfigCategory.IDENTIFY.name]);
+  payload.events = [];
+  const eventsObj = {
+    name: 'Identify User',
+    properties: {
+      dyType: 'identify-v1',
+      hashedEmail:
+        message.traits?.email || message.context?.traits?.email
+          ? sha256(message.traits?.email || message.context?.traits?.email)
+          : null,
```

Comment:
We have to check from the config i.e. if the customer sending hashed data or not.

Reply:
we can also do checking in code itself, if the provided email is already hashed or not. WDYT?

Reply:
Added hashEmail field in config